File: fixtures/scale/scala/large/src/cryptobench/ciphers/playfair/PlayfairKey.scala
package cryptobench.ciphers.playfair

final case class PlayfairKey(keyword: String) {

  def square: String = {
    val sb = new StringBuilder
    for (c <- keyword.toUpperCase.replace("J", "I") + "ABCDEFGHIKLMNOPQRSTUVWXYZ") {
      if (c >= 'A' && c <= 'Z' && c != 'J' && sb.indexOf(c.toString) < 0) sb.append(c)
    }
    sb.toString
  }
}

object PlayfairKey {
  def default(): PlayfairKey = PlayfairKey("MONARCHY")
}

File: fixtures/scale/java/large/src/cryptobench/ciphers/playfair/PlayfairKey.java
package cryptobench.ciphers.playfair;

public class PlayfairKey {
    private final String keyword;

    public PlayfairKey(String keyword) {
        this.keyword = keyword.toUpperCase().replace("J", "I");
    }

    public String square() {
        StringBuilder sb = new StringBuilder();
        String source = keyword + "ABCDEFGHIKLMNOPQRSTUVWXYZ";
        for (char c : source.toCharArray()) {
            if (c >= 'A' && c <= 'Z' && c != 'J' && sb.indexOf(String.valueOf(c)) < 0) {
                sb.append(c);
            }
        }
        return sb.toString();
    }

    public static PlayfairKey defaultKey() {
        return new PlayfairKey("MONARCHY");
    }
}
